oracle/src/main.rs

```diff
@@ -135,69 +77,51 @@ fn get_exponential_backoff() -> ExponentialBackoff {
     }
 }
 
-/// Fetches the exchange rate from CoinGecko or uses the provided default.
-/// This is then converted into the expected format and submitted.
-async fn submit_exchange_rate(
-    parachain_rpc: &InterBtcParachain,
-    url_or_def: UrlOrDefault<FixedU128>,
-    currency_id: CurrencyId,
-    conversion_factor: FixedU128,
-) -> Result<(), Error> {
-    let coingecko_id = currency_id.coingecko_id()?;
-    let exchange_rate = match url_or_def {
-        UrlOrDefault::Url(url) => {
-            // exchange_rate given in BTC so convert after
-            get_exchange_rate_from_coingecko(coingecko_id.clone(), &url).await?
-        }
-        UrlOrDefault::Def(def) => def,
-    };
-
-    let exchange_rate = exchange_rate
-        .checked_mul(&conversion_factor)
-        .ok_or(Error::InvalidExchangeRate)?;
+async fn submit_bitcoin_fees(parachain_rpc: &InterBtcParachain, bitcoin_fee: f64) -> Result<(), Error> {
+    if bitcoin_fee.is_nan() {
+        log::warn!("Not submitting fee estimate");
+        return Ok(());
```

Comment:
Shouldn't this be an error?

Reply:
If there are no Bitcoin price feeds configured this would've returned NaN from the medianization, I've changed the feeder to be more specific and return an Option.